common/queue/src/main/java/org/thingsboard/server/queue/discovery/ZkDiscoveryService.java

```diff
@@ -290,8 +297,32 @@ public void childEvent(CuratorFramework curatorFramework, PathChildrenCacheEvent
         log.debug("Processing [{}] event for [{}]", pathChildrenCacheEvent.getType(), instance.getServiceId());
         switch (pathChildrenCacheEvent.getType()) {
             case CHILD_ADDED:
+                ScheduledFuture<?> task = delayedTasks.remove(instance.getServiceId());
+                if (task != null) {
+                    if (task.cancel(false)) {
+                        log.debug("[{}] Recalculate partitions ignored. Service was restarted in time [{}].",
+                                instance.getServiceId(), instance.getServiceTypesList());
+                    } else {
+                        log.debug("[{}] Going to recalculate partitions. Service was not restarted in time [{}]!",
+                                instance.getServiceId(), instance.getServiceTypesList());
+                        recalculatePartitions();
+                    }
+                } else {
+                    log.debug("[{}] Going to recalculate partitions due to adding new node [{}].",
+                            instance.getServiceId(), instance.getServiceTypesList());
+                    recalculatePartitions();
+                }
+                break;
             case CHILD_REMOVED:
-                recalculatePartitions();
+                ScheduledFuture<?> future = zkExecutorService.schedule(() -> {
+                    log.debug("[{}] Going to recalculate partitions due to removed node [{}]",
```

Comment:
Please, improve logging. For the first time, the logs will be enabled on **debug** level to provide
Please, leave all spam logs on the TRACE level.
For now the debug logs is to spammy

```
2023-07-21 15:02:49,839 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-http-transport-1]
2023-07-21 15:02:49,920 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Ignoring event about current server PathChildrenCacheEvent{type=CHILD_UPDATED, data=ChildData{path='/thingsboard/nodes/0000001554', stat=55836430886,55836458616,....,57,0,55836430886
, data=[10, 19, 116, 98, 45, 115, 110, ... 71, 32, -128, -32, -12, -67, 30, 40, 20, 48, -128, -32, -2, -7, 127]}}
2023-07-21 15:02:56,910 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-mqtt-transport-2]
2023-07-21 15:02:57,487 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-core-1]
2023-07-21 15:03:02,889 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-ie-main-0]
2023-07-21 15:03:04,846 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-lwm2m-transport-0]
2023-07-21 15:03:05,442 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-rule-engine-0]
2023-07-21 15:03:05,445 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-vc-executor-1]
2023-07-21 15:03:06,703 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-vc-executor-0]
2023-07-21 15:03:14,163 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-ie-main-1]
2023-07-21 15:03:17,140 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-ie-main-2]
2023-07-21 15:03:17,940 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-core-2]
2023-07-21 15:03:19,812 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-mqtt-transport-0]
2023-07-21 15:03:23,395 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-rule-engine-1]
2023-07-21 15:03:29,300 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-core-3]
2023-07-21 15:03:30,644 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-core-0]
2023-07-21 15:03:32,277 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-coap-transport-0]
2023-07-21 15:03:33,715 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-http-transport-0]
2023-07-21 15:03:36,360 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-core-4]
2023-07-21 15:03:38,445 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-rule-engine-2]
2023-07-21 15:03:47,979 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-coap-transport-1]
2023-07-21 15:03:48,251 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-mqtt-transport-1]
2023-07-21 15:03:48,754 [Curator-PathChildrenCache-0] DEBUG o.t.s.q.discovery.ZkDiscoveryService - Processing [CHILD_UPDATED] event for [tb-ie-other-0]

```